public/app/routes/routes.tsx

```diff
@@ -434,9 +434,9 @@ export function getAppRoutes(): RouteDescriptor[] {
         () => import(/* webpackChunkName: "SnapshotListPage" */ 'app/features/manage-dashboards/SnapshotListPage')
       ),
     },
-    config.featureToggles.dashboardRestoreUI && {
+    config.featureToggles.dashboardRestore && {
       path: '/dashboard/recently-deleted',
-      roles: () => ['Admin'],
+      roles: () => ['Admin', 'ServerAdmin'],
```

Comment:
Adding this here to complete setting up permissions in a cohesive way.
See https://github.com/grafana/grafana/blob/0573c2f7b62fb4b11bc02c992c0d903f868554b3/public/app/features/browse-dashboards/BrowseDashboardsPage.tsx#L87